src/post_fetch_helpers.jl

```diff
@@ -1,66 +1,22 @@
-# This file is a part of DataDeps.jl. License is MIT.
-
-if Sys.isunix() && Sys.KERNEL != :FreeBSD
-    function unpack_cmd(file,directory,extension,secondary_extension)
-        if ((extension == ".gz" || extension == ".Z") && secondary_extension == ".tar") || extension == ".tgz"
-            return (`tar xzf $file --directory=$directory`)
-        elseif (extension == ".bz2" && secondary_extension == ".tar") || extension == ".tbz"
-            return (`tar xjf $file --directory=$directory`)
-        elseif extension == ".xz" && secondary_extension == ".tar"
-            return pipeline(`unxz -c $file `, `tar xv --directory=$directory`)
-        elseif extension == ".tar"
-            return (`tar xf $file --directory=$directory`)
-        elseif extension == ".zip"
-            return (`unzip -x $file -d $directory`)
-        elseif extension == ".gz"
-            target = joinpath(directory, file)
-            return pipeline(`gzip -k -d $target`)
+function unpack_cmd(file,directory,extension,secondary_extension)
+    if ((extension == ".Z" || extension == ".gz" || extension == ".xz" || extension == ".bz2") && secondary_extension == ".tar") || extension == ".tgz" || extension == ".tbz" || extension == ".zip" || extension== ".gz" || extension == ".7z" || extension == ".tar" || (extension == ".exe" && secondary_extension == ".7z")
```

Comment:
Perhaps break this line in the occurrences of `&&`. It is too long to read.

Reply:
perhaps we could even get rid of this and simplify the whole thing more since we should be able to handle everthing the same?
Perhaps with a remaining special case for `tar`?

Reply:
@oxinabox I'll see if we can get a status code of the `run` command. In case we get a non-zero code, we can notify the user that `There was an error. Please check if the path is correct, if yes, either the file is corrupt or has an unsupported extension.`

@juliohm Alright, I'll break it up if we have to keep them.